### ISSUE TITLE
max() on DateTime with empty result set throws

### ISSUE THREAD

**Issue by igbenic** [simolus3/drift#1898]
I have a table `customsTariffs` with `DateTime? adTimeChg` field, when querying for max of this column:

```dart
final exp = customsTariffs.adTimeChg.max();
final query = selectOnly(customsTariffs)..addColumns([exp]);
return query.map((p0) => p0.read(exp)).getSingleOrNull();
```

the `exp` variable has a type of `Expression<DateTime>`, which is not correct, it should be `Expression<DateTime?>` because the result set can be empty and in that case `exp` returns null which breaks at `p0.read(exp)` with `type 'Null' is not a subtype of type 'DateTime' in type cast`. With this change it works:

```dart
final exp = customsTariffs.adTimeChg.max() as Expression<DateTime?>;
final query = selectOnly(customsTariffs)..addColumns([exp]);
return query.map((p0) => p0.read(exp)).getSingleOrNull();
```

Thank you for all the great work!

**Comment by simolus3** [comment]
Thanks for the report! I've fixed this in 84cf9b82c38062d12a00391a65ff0bb5497aaf12. It's a breaking change unfortunately, so releasing it may take a while since I want to use the opportunity of a major version bump to get more breaking changes done.

Your workaround should work. Alternatively, you can also do `customsTariffs.adTimeChg.max().dartCast<DateTime?>()` which is safer in general but in this case it doesn't make a difference.

**Comment by igbenic** [comment]
Thank you!
